### PR TITLE
OHRM5X-986: Create default Timesheet By System Its Self Feature Added

### PR DESCRIPTION
**Pull request by VageeshOrangeHRM**
This PR Contains following Changes:
1. OHRM5X-986: Create default Timesheet By System Its Self Feature Added.
2. Inside the MyTimesheetController Transaction method added for evaluating the default Timesheet creation.

### CODE REVIEW

File: symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php [hunk 1]
```diff
@@ -787,13 +787,18 @@ public function checkForMatchingTimesheetForCurrentDate($employeeId, $currentDat
...
+    public function hasTimesheetForStartDate(DateTime $date, ?int $employeeNumber = null): bool
     {
         $q = $this->createQueryBuilder(Timesheet::class, 'timesheet');
         $q->andWhere('timesheet.startDate = :date');
         $q->setParameter('date', $date);
+        if (!is_null($employeeNumber)) {
+            $q->andWhere('timesheet.employee = :employeeNumber');
+            $q->setParameter('employeeNumber', $employeeNumber);
+        }
         return $this->getPaginator($q)->count() === 0;
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php:802]:
```suggestion
        return $this->getPaginator($q)->count() > 0;
```
Change the validation API logic as well

File: symfony/plugins/orangehrmTimePlugin/Controller/MyTimesheetController.php [hunk 2]
```diff
@@ -19,18 +19,46 @@
...
+     */
+    private function createDefaultTimesheetIfNotExist(): void
+    {
+        $currentDate = $this->getDateTimeHelper()->getNow();
+        list($startDate) = $this->getTimesheetService()->extractStartDateAndEndDateFromDate($currentDate);
+        $status = $this->getTimesheetService()
+            ->getTimesheetDao()
+            ->hasTimesheetForStartDate(new DateTime($startDate), $this->getAuthUser()->getEmpNumber());
+
+        if ($status) {
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Controller/MyTimesheetController.php:58]:
```suggestion
        if (!$status) {
```

File: symfony/plugins/orangehrmTimePlugin/Service/TimesheetService.php [hunk 3]
```diff
@@ -557,7 +557,7 @@ public function createTimesheetByDate(Timesheet $timesheet, DateTime $date): Tim
      * @param DateTime $date
      * @return array
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Service/TimesheetService.php:558]:
```suggestion
     * @return array e.g array('2021-12-13', '2021-12-19')
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/plugins/orangehrmTimePlugin/Controller/MyTimesheetController.php [hunk 4]
```diff
@@ -19,18 +19,46 @@
...
+    /**
+     * @return void
+     */
+    private function createDefaultTimesheetIfNotExist(): void
+    {
+        $currentDate = $this->getDateTimeHelper()->getNow();
+        list($startDate) = $this->getTimesheetService()->extractStartDateAndEndDateFromDate($currentDate);
+        $status = $this->getTimesheetService()
+            ->getTimesheetDao()
+            ->hasTimesheetForStartDate(new DateTime($startDate), $this->getAuthUser()->getEmpNumber());
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Controller/MyTimesheetController.php:56]:
```suggestion
        $status = $this->getTimesheetService()
            ->hasTimesheetForDate($currentDate, $this->getAuthUser()->getEmpNumber());
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php (hunk 1 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmTimePlugin/Dao/TimesheetDao.php]: done

File: symfony/plugins/orangehrmTimePlugin/Controller/MyTimesheetController.php (hunk 2 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmTimePlugin/Controller/MyTimesheetController.php]: done

File: symfony/plugins/orangehrmTimePlugin/Service/TimesheetService.php (hunk 3 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmTimePlugin/Service/TimesheetService.php]: done

File: symfony/plugins/orangehrmTimePlugin/Controller/MyTimesheetController.php (hunk 4 above)
Comment (VageeshOrangeHRM) [reply symfony/plugins/orangehrmTimePlugin/Controller/MyTimesheetController.php]: done

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**